SpatialGDK/Source/SpatialGDKTests/SpatialGDK/LoadBalancing/LayeredLBStrategy/LayeredLBStrategyTest.cpp

```diff
@@ -0,0 +1,412 @@
+#include "LoadBalancing/LayeredLBStrategy.h"
+#include "Tests/AutomationCommon.h"
+#include "Tests/AutomationEditorCommon.h"
+#include "Tests/TestDefinitions.h"
+#include "Engine/Engine.h"
+#include "GameFramework/GameStateBase.h"
+#include "EngineClasses/SpatialWorldSettings.h"
+#include "LoadBalancing/GridBasedLBStrategy.h"
+#include <vector>
+#include <utility>
+#include "SpatialGDKTests/SpatialGDK/LoadBalancing/GridBasedLBStrategy/TestGridBasedLBStrategy.h"
+#include <string>
+#include "TestLayeredLBStrategy.h"
+#include "GameFramework/DefaultPawn.h"
+#include "Misc/Optional.h"
+
+
+#define LAYEREDLBSTRATEGY_TEST(TestName) \
+	GDK_TEST(Core, ULayeredLBStrategy, TestName)
+
+
+
+namespace
+{
+	ULayeredLBStrategy* Strat;
+	UWorld* TestWorld;
+	TMap<FName, AActor*> TestActors;
+
+	// Copied from AutomationCommon::GetAnyGameWorld().
+	UWorld* GetAnyGameWorld()
+	{
+		UWorld* World = nullptr;
+		const TIndirectArray<FWorldContext>& WorldContexts = GEngine->GetWorldContexts();
+		for (const FWorldContext& Context : WorldContexts)
+		{
+			if ((Context.WorldType == EWorldType::PIE || Context.WorldType == EWorldType::Game)
+				&& (Context.World() != nullptr))
+			{
+				World = Context.World();
+				break;
+			}
+		}
+
+		return World;
+	}
+
+	DEFINE_LATENT_AUTOMATION_COMMAND(FWaitForWorld);
+	bool FWaitForWorld::Update()
+	{
+		TestWorld = GetAnyGameWorld();
+
+		if (TestWorld && TestWorld->AreActorsInitialized())
+		{
+			AGameStateBase* GameState = TestWorld->GetGameState();
+			if (GameState && GameState->HasMatchStarted())
+			{
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	struct StrategyConfig {
+		TSubclassOf<UAbstractLBStrategy> StrategyClass;
+		TSet<TSoftClassPtr<AActor>> TargetActorTypes;
+	};
+
+	DEFINE_LATENT_AUTOMATION_COMMAND_TWO_PARAMETER(FSetupStrategy,
+		std::vector<StrategyConfig>, ComponentStrategyConfigs, TOptional<uint32>, NumVirtualWorkers);
+	bool FSetupStrategy::Update()
+	{
+		ASpatialWorldSettings* WorldSettings = Cast<ASpatialWorldSettings>(TestWorld->GetWorldSettings());
+		WorldSettings->DefaultLayerLoadBalanceStrategy = UGridBasedLBStrategy::StaticClass();
+		WorldSettings->bEnableMultiWorker = true;
+
+		Strat = NewObject<ULayeredLBStrategy>(TestWorld);
+		int StratCounter = 0;
+		for (auto& StrategyConfig : ComponentStrategyConfigs)
+		{
+			auto strat_name = FName{ *FString(std::to_string(StratCounter++).c_str()) };
```

Comment:
prefer to use unreal types, i.e. uint32, then use Unreal functions here as suggested
```suggestion
			auto strat_name = FName{ *FString::FromInt(StratCounter) };
```

Reply:
done